src/contrasts.jl

```diff
@@ -89,6 +89,17 @@ type ContrastsMatrix{C <: AbstractContrasts, T}
     contrasts::C
 end
 
+# only check equality of matrix, termnames, and levels, and that the type is the
+# same for the contrasts (values are irrelevant).  This ensures that the two
+# will behave identically in creating modelmatrix columns
+Base.:(==){C<:AbstractContrasts,T}(a::ContrastsMatrix{C,T}, b::ContrastsMatrix{C,T}) =
```

Comment:
Ah, right. Though it would be clearer with `C1 === C2`, and maybe even shorter since the signature wouldn't need type parameters.